com.kms.cura.dal/src/com/kms/cura/dal/database/AppointmentDatabaseHelper.java

```diff
@@ -178,18 +178,26 @@ private void createAppointment(AppointmentEntity entity) throws SQLException {
 
     }
 
-    public List<AppointmentEntity> updateAppointment(AppointmentEntity entity) throws SQLException, ClassNotFoundException, IOException {
-        List<AppointmentEntity> patientAppts;
+    public List<AppointmentEntity> updateAppointment(AppointmentEntity entity, boolean patient) throws SQLException, ClassNotFoundException, IOException {
+        List<AppointmentEntity> listAppts = null;
+        AppointmentEntity search = null;
         PreparedStatement stmt = null;
         try {
             con.setAutoCommit(false);
             stmt = getUpdateAppointmentQuery(entity);
             stmt.executeUpdate();
-            AppointmentEntity search = new AppointmentEntity(entity.getPatientUserEntity(), null, null, null, null,
+            if (patient){
+            	search = new AppointmentEntity(entity.getPatientUserEntity(), null, null, null, null,
```

Comment:
Use a local variable to save entity.getPatientUserEntity() or entity.getDoctorUserEntity()